src/lib/serialization/schema.ts

```diff
@@ -0,0 +1,255 @@
+type _ModelToObject<T> =
+    // Reflections
+    T extends M.ReflectionGroup ? JSONOutput.ReflectionGroup :
+    T extends M.ReflectionCategory ? JSONOutput.ReflectionCategory :
+    T extends M.SignatureReflection ? JSONOutput.SignatureReflection :
+    T extends M.ParameterReflection ? JSONOutput.ParameterReflection :
+    T extends M.DeclarationReflection ? JSONOutput.DeclarationReflection | JSONOutput.ReflectionPointer :
+    T extends M.TypeParameterReflection ? JSONOutput.TypeParameterReflection :
+    T extends M.ProjectReflection ? JSONOutput.ProjectReflection :
+    T extends M.ContainerReflection ? JSONOutput.ContainerReflection :
+    T extends M.Reflection ? JSONOutput.Reflection :
+    // Types
+    T extends M.ArrayType ? JSONOutput.ArrayType :
+    T extends M.IntersectionType ? JSONOutput.IntersectionType :
+    T extends M.IntrinsicType ? JSONOutput.IntrinsicType :
+    T extends M.ReferenceType ? JSONOutput.ReferenceType :
+    T extends M.ReflectionType ? JSONOutput.ReflectionType :
+    T extends M.StringLiteralType ? JSONOutput.StringLiteralType :
+    T extends M.TupleType ? JSONOutput.TupleType :
+    T extends M.UnknownType ? JSONOutput.UnknownType :
+    T extends M.Type ? JSONOutput.SomeType : // Technically AbstractType, but the union is more useful
+    // Miscellaneous
+    T extends M.Comment ? JSONOutput.Comment :
+    T extends M.CommentTag ? JSONOutput.CommentTag :
+    T extends DecoratorWrapper ? JSONOutput.Decorator :
+    T extends SourceReferenceWrapper ? JSONOutput.SourceReference :
+    never;
+
+/**
+ * Contains interfaces which describe the JSON output. Each interface is related to a specific type of serializer.
+ *
+ * ## Plugins
+ * Plugins which modify the serialization process can use declaration merging
+ * to add custom properties to the exported interfaces.
+ * For example, if your custom serializer adds a property to all [[Reflection]] objects:
+ * ```ts
+ * declare module 'typedoc/dist/lib/serialization/schema' {
+ *     export namespace JSONOutput {
+ *         export interface AbstractReflection {
+ *             myCustomProp: boolean
+ *         }
+ *     }
+ * }
+ * ```
+ *
+ * If a plugin defines a new Model type, [[ModelToObject]] will not pick up the serializer type.
+ * To fix this, use declaration merging to augment the [[Serializer]] class.
+ * ```ts
+ * declare module 'typedoc/dist/lib/serialization/serializer' {
+ *     export interface Serializer {
+ *         toObject(value: CustomModel, obj?: Partial<CustomModel>): CustomOutput
+ *     }
+ * }
+ * ```
+ *
+ * For documentation on the JSON output properties, view the corresponding model.
+ */
+export namespace JSONOutput {
+
+    // Reflections
+
+    export interface ReflectionGroup {
+        title: M.ReflectionGroup['title'];
+        kind: M.ReflectionGroup['kind'];
+        children?: M.ReflectionGroup['children'][number]['id'][];
+        categories?: ModelToObject<M.ReflectionGroup['categories']>;
```

Comment:
Yes, it absolutely makes sense :+1: Will do.

Reply:
Coming back to this, I'm not sure it does make sense. It's still an option, but having all the types in one place lets us avoid importing the `S` helper I introduced to resolve another of your comments everywhere (and thus keeps that particular bit of ugly out of the rest of of the code)

On the other hand, it would be nice to have the types closer to the code that produces values of that type...